delete legacy component library scss

mbylstra opened the pull request:
# Objective
<!-- Describe what this change achieves, and the details of how it works. -->

# Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

# Screenshots (if appropriate)

# Checklist
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] If this contains visual changes, has it been reviewed by a designer?
- [ ] If this introduces either a new component or a breaking change to an existing component, has it been reviewed by an advocate? (Ask in #prod_design_systems)
- [ ] I have considered likely risks of these changes and got someone else to QA as appropriate
- [ ] I have or will communicate these changes to the front end practice

# Additional Considerations
- Does there need to be right-to-left (RTL) options for localization and internationalization?
- Has the test suite been updated?
- Have you done cross-browser testing for our [supported browsers](https://academy.cultureamp.com/hc/en-us/articles/204539569-Supported-browsers-for-Participants)?
- Have you updated any relevant documentation or left useful comments in the code?
- Have you reviewed Culture Amp's Web Accessibility guide [Current Compliance, Going Forward, Statement and Answers for Customers](https://cultureamp.atlassian.net/wiki/spaces/Prod/pages/428572998/Web+Accessibility)?
- If this contains substantial visual changes, especially far reaching ones, have you unlocked the Chromatic step in the pipeline?
- Have Storybook stories been updated?

**If you're new to Kaizen, please ask #prod_design_systems to set up an onboarding session to get you up to speed.** If you have an urgent PR to merge before that happens, it is safest to ask in #prod_design_systems to have a design system advocate review the PR to catch any issues.

mbylstra commented on package.json:
```diff
@@ -16,6 +16,7 @@
     "eslint": "eslint . -c .eslintrc.js --ext .ts,.tsx",
     "prettier": "prettier --check '**/*'",
     "test": "jest",
+    "jest:debug": "node --inspect-brk node_modules/.bin/jest --runInBand --no-cache",
```
This is just something helpful I added when trying to debug a jest test using the Chrome debugger

mbylstra commented on packages/deprecated-component-library-helpers/styles/color.scss:
```diff
@@ -30,25 +30,25 @@ $ca-palette-peach: #f3786d;
 $ca-palette-yuzu: #ffce1e;
```
`packages/deprecated-component-library-helpers/styles/color.scss` had drifted a little out of sync with  `component-library/styles/color.scss`. This fixes that (before `component-library/styles/color.scss` is deleted)

mbylstra commented on packages/stylelint-plugin/src/index.ts:
```diff
@@ -21,6 +21,10 @@ const kaizenStylelintRules = {
   ],
+  "kaizen/use-deprecated-component-library-helpers-scss-imports": [
+    true,
+    { severity: "error", disableFixing: false },
```
The severity is "error" because this PR also introduces a breaking change where the old imports dissapear (if the consumer doens't run `--fix` then their codebase won't even compile)

mbylstra commented on packages/stylelint-plugin/src/stylelintPlugin.ts:
```diff
@@ -47,7 +49,7 @@ export default rules.map(rule =>
               ruleName: `kaizen/${rule.name}`,
               message: `${message}${
-                autofixAvailable ? "(autofix avaialable)" : ""
+                autofixAvailable ? " (autofix available)" : ""
```
😃

mbylstra commented on packages/stylelint-plugin/test.scss:
```diff
@@ -5,7 +5,7 @@
 @import "././another.scss";
 
-@import "~@kaizen/component-library/styles/layout";
+@import "~@kaizen/deprecated-component-library-helpers/styles/layout";
```
There was just one place in this repo that needed to be fixed. This commit was created with `stylelint --fix`

mbylstra commented on packages/component-library/stories/ToastNotification.stories.scss:
```diff
@@ -1,4 +1,4 @@
-@import "~@kaizen/component-library/styles/type";
+@import "~@kaizen/deprecated-component-library-helpers/styles/type";
```
This is one I had to manually. I think it might be because `ToastNotification.stories.scss` is not picked up by `stylelint **/*.scss`, but I'm not completely sure.

ActuallyACat commented on packages/component-library/styles/color.scss:
```diff
@@ -1,63 +0,0 @@
-@import "~@kaizen/design-tokens/sass/color-vars";
```
This is so good to see! 👏🏻

ActuallyACat replied:
As excited as this makes me, I do wonder if it's _potentially_ a little heavy-handed. Could we release the Stylelint rule first before making this breaking change?

mbylstra replied:
That's a really good point. My thinking was just that we couldn't release the breaking change without a Stylelint rule, but there's no reason we couldn't release the stylelint plugin first. I'd like us to get to the point where we're so confident with automatic breaking change updates that it's not even an issue, but yeah, probably best to err on the safer side at this point!

mbylstra replied:
The other upside to just releasing the stylelint rule is that even if consumers don't want to upgrade right way, consumers will stop getting thousands of annoying scss warnings.

mbylstra replied:
I have split out the actual breaking change into a separate  PR (https://github.com/cultureamp/kaizen-design-system/pull/1950), which I will rebase and fix up once this is merged.

ActuallyACat commented on packages/deprecated-component-library-helpers/styles/color.scss:
```diff
@@ -30,25 +30,25 @@ $ca-palette-peach: #f3786d;
-$ca-palette-stone: add-tint($ca-palette-ink, 97%);
-$ca-palette-ash: add-tint($ca-palette-ink, 94%);
-$ca-palette-positive: add-tint($ca-palette-seedling, 30%);
+$ca-palette-stone: #f9f9f9; // previously add-tint($ca-palette-ink, 97%);
```
Just a thought - I wonder if the next phase of this upgrade is to make all of these variables warnings in stylelint?

mbylstra replied:
Yeah, that would be good. Shall we make an issue for this? Were you thinking any hardcoded css color that isn't using a token (it would have to ignore the token files of course!)? or just ones with "previously add-tint" in the comments?

ActuallyACat commented on packages/stylelint-plugin/src/rules/use-deprecated-component-library-helpers-scss-imports.ts:
```diff
@@ -0,0 +1,85 @@
+  },
+]
+
+export const useDeprecatedComponentLibraryHelpersScssImports: RuleDefinition = {
```
What this rule is and does should be described in the README.

mbylstra replied:
done

ActuallyACat reviewed (commented):
Overall this is looking really great, nice one 👏🏻 

I noticed there's some missing documentation. I have a concern that releasing the Stylelint fix _and_ making the breaking change at the same time is a little too heavy-handed. To be on the safe side we might want to consider releasing the Stylelint plugin first, getting developers to run it in their repos, then make the breaking change. My mind can be changed on this if the owners of the monoliths don't think it's a concern.

mbylstra commented:
Thanks for the review! I'll add docs on the stylelint rule and slit the breaking change to a separate PR. I might not get to this until next monday.

mbylstra commented on packages/stylelint-plugin/README.md:
```diff
@@ -31,15 +31,16 @@ These rules are included by default:
 - `kaizen/prefer-color-tokens` - Detects inlined color values (in any kind of form such as hex, rgba, hsl, hwb etc) and recommends and replaces Kaizen tokens for them if one is found with the same value. Only non-deprecated tokens will be recommended.
 
-- `kaizen/no-tokens-in-variables` - Prevent and fix the usage of Kaizen design tokens in Sass or Less variables. This does not include design tokens in variables that have been imported into the current stylesheet. For example, `$card-bg: $kz-color-wisteria-800; .card { background-color: $card-bg; }` would be disallowed and would be autofixed to `background-color: $kz-color-wisteria-800;`. 
+- `kaizen/no-tokens-in-variables` - Prevent and fix the usage of Kaizen design tokens in Sass or Less variables. This does not include design tokens in variables that have been imported into the current stylesheet. For example, `$card-bg: $kz-color-wisteria-800; .card { background-color: $card-bg; }` would be disallowed and would be autofixed to `background-color: $kz-color-wisteria-800;`.
```
It looks like these changes here are just whitespace difference due to a markdown formatter I must have been running in my editor.

mbylstra commented:
@allybeaton I've split out the breaking change and I've added the new rule to the README, if you could give this a re-review when you get a chance. Thanks!
